src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ProvideStreamMemoryAsyncOverridesTests.cs

```diff
@@ -0,0 +1,1476 @@
+            {
+                TestCode = code,
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                ExpectedDiagnostics =
+                {
+                    VerifyCS.Diagnostic(Rule)
+                        .WithLocation(0)
+                        .WithLocation(1)
+                        .WithArguments("River", displayArrayMethod, displayMemoryMethod),
+                }
+            };
+            return test.RunAsync();
+        }
+
+        [Theory]
+        [InlineData(VBReadAsyncArray, VBDisplayReadAsyncArray, VBDisplayReadAsyncMemory)]
+        [InlineData(VBWriteAsyncArray, VBDisplayWriteAsyncArray, VBDisplayWriteAsyncMemory)]
+        public Task SingleArrayOverride_MultiplePartialsInSameFile_ReportsAllLocations_VB(string arrayMethod, string displayArrayMethod, string displayMemoryMethod)
+        {
+            string code = $@"
+{VBUsings}
+Namespace Testopolis
+    Partial Public Class {{|#0:River|}} : Inherits Stream
+        {VBAbstractMembers}
+        {arrayMethod}
+    End Class
+End Namespace
+
+Namespace Testopolis
+    Partial Class {{|#1:River|}}
+    End Class
+End Namespace";
+
+            var test = new VerifyVB.Test
+            {
+                TestCode = code,
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                ExpectedDiagnostics =
+                {
+                    VerifyVB.Diagnostic(Rule)
+                        .WithLocation(0)
+                        .WithLocation(1)
+                        .WithArguments("River", displayArrayMethod, displayMemoryMethod)
+                }
+            };
+            return test.RunAsync();
+        }
+
+        [Theory]
+        [InlineData(CSReadAsyncArray, CSDisplayReadAsyncArray, CSDisplayReadAsyncMemory)]
+        [InlineData(CSWriteAsyncArray, CSDisplayWriteAsyncArray, CSDisplayWriteAsyncMemory)]
+        public Task SingleArrayOverride_MultiplePartialsInSeparateFiles_ReportsAllLocations_CS(string arrayMethod, string displayArrayMethod, string displayMemoryMethod)
+        {
+            string fooSource = $@"
+{CSUsings}
+namespace Testopolis
+{{
+    public partial class {{|#0:River|}} : Stream
+    {{
+        {CSAbstractMembers}
+    }}
+}}";
+            string barSource = $@"
+{CSUsings}
+namespace Testopolis
+{{
+    partial class {{|#1:River|}}
+    {{
+        {arrayMethod}
+    }}
+}}";
+            string bazSource = $@"
+{CSUsings}
+namespace Testopolis
+{{
+    partial class {{|#2:River|}}
+    {{
+    }}
+}}";
+            var test = new VerifyCS.Test
+            {
+                TestState = { Sources = { fooSource, barSource, bazSource } },
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50,
+                ExpectedDiagnostics =
+                {
+                    VerifyCS.Diagnostic(Rule)
+                        .WithLocation(0)
+                        .WithLocation(1)
+                        .WithLocation(2)
```

Comment:
Same comment here.

Reply:
Same as my previous comment, but this time we are checking whether all locations are reported when the partials are in separate files.